rationale.md

```diff
@@ -383,42 +427,192 @@ async create_query_task(
+Using this type for the parameter also helps document the expected format for the list submitted to the API endpoint.
+
+Both `DelimArray` and `DelimSequence` accept optional overrides for:
+
+- `separator`, which default to a comma. This can be overridden with pipe, space, tab, tilde or whatever string separator might be required.
+
+- `prefix`, which defaults to an empty string.
+
+- `suffix`, which defaults to an empty string.
+
+### Binary response handling
+
+The popular request libraries currently used by the RTLs for both Python and the Node-based Typescript SDK treat response content as string by default, with binary mode as an opt-in setting. This results in the download of binary content like images or PDFs from SDK methods to be corrupted.
+
+Until we've [resolved this issue](#binary-response-detection) with replacement HTTP request methods, the code generator will assist with binary responses in the following ways (the provided examples in Typescript).
+
+#### Response can be either binary and string
+
+A **mixed mode** response that can be either binary or string (e.g. text or json) has a note placed into the method header.
+
+```typescript
+/**
+  * POST /queries/run/{result_format} -> string}
+  *
+  * **Note**: Binary content may be returned by this method. Add `binaryMode` to the `options` parameter to correctly receive binary content
+  */
+async run_inline_query(
+  request: Partial<IRequestrun_inline_query>,
+  options?: Partial<ITransportSettings>,
+) {
+  return this.post<string, IError>(
+    encodeURI(`/queries/run/${request.result_format}`),
+    {
+      limit: request.limit,
+      apply_formatting: request.apply_formatting,
+      apply_vis: request.apply_vis,
+      cache: request.cache,
+      image_width: request.image_width,
+      image_height: request.image_height,
+      generate_drill_links: request.generate_drill_links,
+      force_production: request.force_production,
+      cache_only: request.cache_only,
+      path_prefix: request.path_prefix,
+      rebuild_pdts: request.rebuild_pdts,
+      server_table_calcs: request.server_table_calcs,
+    },
+    request.body,
+    options,
+  )
+}
+```
+
+#### Binary responses
+
+A method that only has binary responses will have the request configured for binary inside the generated SDK method. Images and PDFs are considered binary.
+
+```typescript
+/**
+  * GET /render_tasks/{render_task_id}/results -> string}
+  *
+  * **Note**: Binary content is returned by this method.
+  */
+async render_task_results(
+  /**
+    * @param {string} render_task_id Id of render task
+    */
+  render_task_id: string,
+  options?: Partial<ITransportSettings>,
+) {
+  options = { options, ...binaryMode }
+  return this.get<string, IError>(
```

Comment:
Looks like this is returning a string?  Shouldn't binary be returned as an array of byte?

Reply:
"A string is a series of bytes" https://kevin.burke.dev/kevin/node-js-string-encoding/